proxy/src/main/java/com/wavefront/agent/sampler/EbpfSampler.java

```diff
@@ -0,0 +1,184 @@
+package com.wavefront.agent.sampler;
+
+import static com.wavefront.common.TraceConstants.PARENT_KEY;
+import static com.wavefront.sdk.common.Constants.*;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.wavefront.sdk.entities.tracing.sampling.RateSampler;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.LongAccumulator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.StringUtils;
+import wavefront.report.Annotation;
+import wavefront.report.Span;
+
+/**
+ * This Sampler is for Ebpf traffic focusing on not losing a single edge due to sampling in
+ * application topology. It uses a dynamic sampling rate and prioritizes less frequent edges, error
+ * spans and long spans
+ */
+public class EbpfSampler implements PreferedSampler {
+  private static final String OTEL_SCOPE_KEY = "otel.scope.name";
+  private static final String OTEL_SCOPE_VALUE_PIXIE = "pixie";
```

Comment:
The span tag to identify auto-instrumentation/ebpf data has changed to `origin`, and the value will be `opapps-auto`.

@xuranchen [added an `origin` constant](https://github.com/wavefrontHQ/wavefront-sdk-java/pull/275) to the Java SDK that's available in v3.2.0+. This is where we're setting the span tag in the PxL script: https://github.com/wavefrontHQ/observability-for-kubernetes/blob/4f3f644f59e2b8dfd9e94feca020e5fe9cbdd716/operator/hack/autoinstrumentation/spans.pxl#L171